alpha/veneer/basic/basic.go

```diff
@@ -0,0 +1,101 @@
+package basic
+
+import (
+	"context"
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-registry/alpha/action"
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"github.com/operator-framework/operator-registry/alpha/property"
+	"github.com/operator-framework/operator-registry/pkg/image"
+)
+
+var ErrNotAllowed = errors.New("not allowed")
+var ErrCannotIngestDirectory = errors.New("cannot ingest veneers by directory reference")
+
+type BasicVeneer struct {
+	Registry image.Registry
+}
+
+func (v BasicVeneer) Ingest(ctx context.Context, ref string) (*declcfg.DeclarativeConfig, error) {
+	// only taking first argument as file
+	stat, serr := os.Stat(ref)
+	if serr == nil {
+		if stat.IsDir() {
+			return nil, ErrCannotIngestDirectory // BUGBUG enhance to walk and eat all files in turn, heeding ignore file
+		} else {
+			return v.renderFile(ctx, ref)
+		}
+	}
+	return nil, serr
```

Comment:
In Go, its idiomatic to keep indents to a minimum, so you see bail out kind of patterns such that happy path typically flows to the end of the function. I'd suggest:
```suggestion
	stat, err := os.Stat(ref)
	if err != nil {
		return nil, serr
	}
	if stat.IsDir() {
		return nil, ErrCannotIngestDirectory // BUGBUG enhance to walk and eat all files in turn, heeding ignore file
	}
	return v.renderFile(ctx, ref)
```